crates/rust-analyzer/src/flycheck.rs

```diff
@@ -419,6 +420,7 @@ impl FlycheckActor {
             self.command_receiver.take();
             self.report_progress(Progress::DidCancel);
             self.diagnostics_cleared_for.clear();
+            self.diagnostics_received = false;
```

Comment:
The bug for things not clearing is here, this line and the above should not be gated by the surrounding if let